features/lwipstack/LWIPStack.cpp

```diff
@@ -508,6 +509,24 @@ nsapi_error_t LWIP::setsockopt(nsapi_socket_t handle, int level, int optname, co
     struct mbed_lwip_socket *s = (struct mbed_lwip_socket *)handle;
 
     switch (optname) {
+    	case NSAPI_BIND_TO_DEVICE:
+             if (optlen != sizeof(char[2])) {
```

Comment:
I think we do not want to fix the length. There should be maximum length and name is ok if it is shorter.

Reply:
Done